cardano-cli/src/Cardano/CLI/Shelley/Run/Governance.hs

```diff
@@ -177,3 +221,152 @@ runGovernanceUpdateProposal (OutputFile upFile) eNo genVerKeyFiles upPprams mCos
+runGovernanceCreatePoll govPollQuestion govPollAnswers govPollNonce out = do
+  let poll = GovernancePoll{ govPollQuestion, govPollAnswers, govPollNonce }
+
+  let description = fromString $ "An on-chain poll for SPOs: " <> Text.unpack govPollQuestion
+  firstExceptT ShelleyGovernanceCmdTextEnvWriteError . newExceptT $
+    writeFileTextEnvelope (unOutputFile out) (Just description) poll
+
+  let metadata = asTxMetadata poll
+        & metadataToJson TxMetadataJsonDetailedSchema
+
+  let outPath = unOutputFile out
+        & Text.encodeUtf8 . Text.pack
+
+  liftIO $ do
+    BSC.hPutStrLn stderr
+      "Poll created successfully.\n\
+      \Please submit a transaction using the resulting metadata.\n"
+    BSC.hPutStrLn stdout (prettyPrintJSON metadata)
+    BSC.hPutStrLn stderr $
+      "\n\
+      \Hint (1): Use '--json-metadata-detailed-schema' and '--metadata-json-file' \
+      \from the build or build-raw commands.\n\
+      \Hint (2): You can redirect the standard output of this command to a JSON \
+      \file to capture metadata.\n\n\
+      \Note: A serialized version of the poll suitable for sharing with \
+      \participants has been generated at '" <> outPath <> "'."
+
+runGovernanceAnswerPoll
+  :: FilePath
+  -> SigningKeyFile  -- VRF or Ed25519 cold key
+  -> Maybe Word -- Answer index
+  -> ExceptT ShelleyGovernanceCmdError IO ()
+runGovernanceAnswerPoll pollFile skFile maybeChoice = do
+  poll <- firstExceptT ShelleyGovernanceCmdTextEnvReadError . newExceptT $
+    readFileTextEnvelope AsGovernancePoll pollFile
+
+  credentials <- readVRFOrColdSigningKeyFile skFile
+
+  choice <- case maybeChoice of
+    Nothing -> do
+      askInteractively poll
+    Just ix -> do
+      validateChoice poll ix
+      liftIO $ BSC.hPutStrLn stderr $ Text.encodeUtf8 $ Text.intercalate "\n"
+        [ govPollQuestion poll
+        , "→ " <> (govPollAnswers poll !! fromIntegral ix)
+        , ""
+        ]
+      pure ix
+
+  let pollAnswer = GovernancePollAnswer
+        { govAnsPoll = hashGovernancePoll poll
+        , govAnsChoice = choice
+        }
+  let witness = pollAnswer `signPollAnswerWith` credentials
+
+  let metadata =
+        mergeTransactionMetadata
+          ( \l r -> case (l, r) of
+              (TxMetaMap xs, TxMetaMap ys) -> TxMetaMap (xs <> ys)
+              _ -> error "unreachable"
+          )
+          (asTxMetadata pollAnswer)
+          (asTxMetadata witness)
+        & metadataToJson TxMetadataJsonDetailedSchema
+
+  liftIO $ do
+    BSC.hPutStrLn stderr
+      "Poll answer created successfully.\n\
+      \Please submit a transaction using the resulting metadata.\n"
+    BSC.hPutStrLn stdout (prettyPrintJSON metadata)
+    BSC.hPutStrLn stderr
+      "\n\
+      \Hint (1): Use '--json-metadata-detailed-schema' and '--metadata-json-file' \
+      \from the build or build-raw commands.\n\
+      \Hint (2): You can redirect the standard output of this command to a JSON \
+      \file to capture metadata."
```

Comment:
What are the trade-offs for writing interactive output to `stderr` and JSON to `stdout` vs writing interactive output to `stdout` and JSON directly to a file?

Reply:
`stdout` = relevant content that should be structured to be piped into other tools or send to files
`stderr` = debug information useful to communicate context and details to users

A command-line that outputs interactive debug information on stdout is arguably doing something wrong; unless it's the main terminal output of the application (e.g. printing structured logs on stdout). 

Then, why stdout rather than a file? Because I find the UX a lot better that way. Command lines with too many params are arguably hard to process; and using file as the medium of exchanges makes it harder / prevent piping into other tools easily. When printing structured results on stdout, one can always redirect the output to a file should they want it; so IMO stdout should always be the default; and files used only when necessary. 

Here I am only using an output file in the case of `create-poll` as a "build artifact". It allows to produce two outputs with distinct purpose; the file is _meant_ to be shared as file, and thus it makes sense to treat it as such from the CLI as well. It also makes it clearer for users (even though that's going to be only super users here) what is meant to be shared and what is metadata.